data/binding/printf.go

```diff
@@ -0,0 +1,250 @@
+package binding
+
+import (
+	"fmt"
+)
+
+type sprintfString struct {
+	String
+
+	format string
+	source []DataItem
+	err    error
+}
+
+// NewSprintf returns a String binding that format its content using the
+// format string and the provide additional parameter that must be other
+// data bindings.
+//
+// Since: 2.2
+func NewSprintf(format string, b ...DataItem) (String, error) {
+	ret := &sprintfString{String: NewString(), format: format}
+
+	for _, value := range b {
+		value.AddListener(ret)
+
+		ret.source = append(ret.source, value)
+	}
+
+	return ret, nil
+}
+
+func (s *sprintfString) DataChanged() {
+	data := make([]interface{}, 0)
+
+	s.err = nil
+	for _, value := range s.source {
+		switch x := value.(type) {
+		case Bool:
+			{
+				b, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, b)
+				break
+			}
```

Comment:
```suggestion
		b, err := x.Get()
		if err != nil {
			s.err = err
			return
		}

		data = append(data, b)
		break
```
I'm not against it per say, but is there any specific reason that you are launching a new scope here?

Reply:
It allow me to use variable that are only valid inside that block and I do not have to start inventing a lot of other variable name that make things, I find, more confusing.

Reply:
Case is a new scope without the braces. And you don’t need break either, that is the default in Go (use `fallthrough` to do the C default)

Reply:
Oh, I didn't know. Thanks, corrected now.